devops/Dockerfile.worker

```diff
@@ -1,46 +1,92 @@
+    python3-dev \
+    ffmpeg \
+    unzip \
+    wget \
+    libc6-dev \
+    libtbb-dev\
+    libcgal-dev
 
 WORKDIR /opt
 
-RUN git clone --depth 1 https://github.com/m-schuetz/LAStools.git && cd LAStools/LASzip && mkdir build && cd build && \
-cmake -DCMAKE_BUILD_TYPE=Release .. && make && make install && ldconfig
-
-RUN git clone -b develop https://github.com/potree/PotreeConverter.git && cd PotreeConverter && git checkout 685ef56a7864ea2a9781b2ab61580f11f0983d29 && \
+# Install PotreeConverter
+# c2328c4 is v2.1.1 and some additional fixes
+RUN git clone -b develop https://github.com/potree/PotreeConverter.git && cd PotreeConverter && git checkout c2328c4 && \
 mkdir build && cd build && \
-cmake -DCMAKE_BUILD_TYPE=Release -DLASZIP_INCLUDE_DIRS=/opt/LAStools/LASzip/dll/ -DLASZIP_LIBRARY=/usr/local/lib/liblaszip.so .. && \
-make && make install && cp -r /opt/PotreeConverter/PotreeConverter/resources /resources
-ADD devops/misc/potree/page_template /resources/page_template
+cmake .. -DCMAKE_BUILD_TYPE=Release && \
+make
 
-RUN update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 2
+# Setup our page template for PotreeConverter
+ADD devops/misc/potree/page_template/nsf_logo.png /opt/PotreeConverter/build/resources/page_template/
+ADD devops/misc/potree/page_template/nsf_logo_snippet.txt /tmp/
+#  - add nsf logo
+RUN sed -i '/<body>/r /tmp/nsf_logo_snippet.txt' /opt/PotreeConverter/build/resources/page_template/viewer_template.html
+#  - remove reference to background image
+RUN sed -i 's/style="[^"]*background-image:[^"]*"//' /opt/PotreeConverter/build/resources/page_template/viewer_template.html
 
-RUN pip3 install --upgrade pip
 
-ENV POETRY_VERSION=1.8.3
-ENV POETRY_HOME=/opt/poetry
-ENV PATH="$POETRY_HOME/bin:$PATH"
-RUN curl -sSL https://install.python-poetry.org | python3 -
-RUN poetry config virtualenvs.create false
-COPY devops/pyproject.toml devops/poetry.lock ./
+# Install Miniforge for our Python environment (provides easier PDAL installation)
+RUN wget -q -O miniforge.sh https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-$(uname -m).sh && \
+    sh miniforge.sh -b -p /opt/conda && \
+    rm miniforge.sh
+ENV PATH="/opt/conda/bin:${PATH}"
 
-WORKDIR /opt
+# Create a conda environment with Python 3.9 and activate it
+RUN conda create -n py39env python=3.9 -y
+SHELL ["conda", "run", "-n", "py39env", "/bin/bash", "-c"]
 
-# install geos into condo the base pdal image is using
-RUN conda install setuptools geos -y -n base
+# Install PDAL using conda
+RUN conda install -c conda-forge pdal -y
 
+# Install needed python packages using poetry
+RUN pip install poetry==1.8.3
+RUN poetry config virtualenvs.create false
+COPY devops/pyproject.toml devops/poetry.lock ./
 RUN poetry install
 
-ENV PYTHONPATH "${PYTHONPATH}:/app"
-
-WORKDIR /
+# Populate image with geoapi and set PYTHONPATH
 RUN mkdir app
 COPY geoapi /app/geoapi
 WORKDIR /app/geoapi
+ENV PYTHONPATH "${PYTHONPATH}:/app"
```

Comment:
This throws a docker warning for using the legacy variable format, instead try: 
`ENV PYTHONPATH="${PYTHONPATH}:/app"`

Reply:
Same tweak goes for line 22 of the main Dockerfile (not the workers one), but its not in this PR...

Reply:
thanks.